### PR TITLE
Assignment Review

### PR DESCRIPTION
**Pull request by sudharsan-munusami**
Once the REVIEW_BRANCH is reviewed and approved, it can be merged.

### CODE REVIEW

**Review by leninkumar-sv-tiger — CHANGES_REQUESTED** [review]
1. 	4b-2: Open and activate a virtual environment (name:htassign)
	4b-3: Instal the requirements file
	- Can you please include the steps used for creating virtual environment ?
	
2. 4b-4: Install awscli and configure using the s3 access ID and Key
	- Please mention which version of awscli you have installed ?
	- Briefly mention the installation steps
	
3. 4b-5: Run the application using python app.py
	- Steps or Commands to install the python application ?
	
4. The security group of the instance is configured as below (allowing through port:8085)
	- In the inbould security rules, I could see other port mappings are used.
	  Are they needed ? Can the application be accessed even if it is accessed from outside ?
	- Ideally 192.168.43.11/32 and 106.197.40.102/32 should only be needed correct ?
	
5. Billing information post completion:
	- Thanks for pasting this screenshot.
	- Ideally the cost-report/free-tier usages will take 24 hrs I guess to get reflected.
	- Can you also please share the percentage of free-tier-usage that the EC2 has incurred ?

**Comment by sudharsan-munusami** [comment]
Hi Lenin,
I have made the changes as requested in both "Main" branch and "REVIEW_BRANCH". Kindly go through the changes and let me know incase of any queries.

Thank you for the feedback!

File: README.md [hunk 1]
```diff
@@ -24,17 +24,26 @@ Procedure:
...
-  
-        ![image](https://user-images.githubusercontent.com/82586609/115119361-93160680-9fc5-11eb-99ab-f3925441f530.png)  
-  
-        ![image](https://user-images.githubusercontent.com/82586609/115119367-a0cb8c00-9fc5-11eb-8050-5dcf67bd17d6.png)  
-  
-        ![image](https://user-images.githubusercontent.com/82586609/115119439-fa33bb00-9fc5-11eb-93c4-b27057a909c8.png)  
-  
+                Before running, check if python is installed by running "python --version"  
+                If not installed, try the below commands in order:  
+                sudo apt update
```
Comment (leninkumar-sv-tiger) [inline README.md:41]: Whenever you are putting in code, please use like this
```bash
sudo apt update
```
This is a better way to mention steps.

Comment (sudharsan-munusami) [reply README.md:41]: Got it!

**Review by leninkumar-sv-tiger — COMMENTED** [review]
- Ideally only when the PR is approved the branch has to be merged to master. It is fine for this assignment
- s3_client = boto3.client('s3'), from s3_demo.py, is used in all the methods. Instead this can be mapped only once in app.py and then can be passed to the methods. This is just an improvement.

**Review by leninkumar-sv-tiger — APPROVED** [review]
Approved. 
Please follow these best practices in the upcoming assignments.
Please merge it for the time being.

**Comment by leninkumar-sv-tiger** [comment]
The submission looks cool

**Comment by sudharsan-munusami** [comment]
Sure Lenin, will incorporate all the suggestions you gave in coming assignments.

Thanks a lot for all the valuable feedback!

**Pull request merged**